src/TimerDialog.cpp

```diff
@@ -0,0 +1,312 @@
+    else
+        noteEdit->setText(noteEdit->text() + " and " + n);
+}
+
+void TimerDialog::setBoil(BoilTime *bt)
+{
+    boilTime = bt;
+}
+
+int TimerDialog::getTime()
+{
+    /*invert to return addition time not time to addition.
+    getTime() and getNote() are used for timer checks when
+    generating timers from recipes
+    */
+    return boilTime->getTime() - time;
+}
+
+QString TimerDialog::getNote()
+{
+    return noteEdit->text();
+}
+
+void TimerDialog::updateTime()
+{
+    timeLCD->display(timeToString(time));
+}
+
+QString TimerDialog::timeToString(int t)
+{
+    unsigned int seconds = t;
+    unsigned int minutes = 0;
+    unsigned int hours = 0;
+   if (t == 0)
+       return "00:00:00";
+   if (t > 59){
+       seconds = t%60;
+       minutes = t/60;
+       if (minutes > 59){
+          hours = minutes/60;
+          minutes = minutes%60;
+       }
+   }
+   QString secStr, minStr, hourStr;
+   if (seconds < 10)
+       secStr = "0" + QString::number(seconds);
+   else
+       secStr = QString::number(seconds);
+   if (minutes <10)
+       minStr = "0" + QString::number(minutes);
+   else
+       minStr = QString::number(minutes);
+   if (hours < 10)
+       hourStr = "0" + QString::number(hours);
+   else
+       hourStr = QString::number(hours);
+   return hourStr + ":" + minStr + ":" + secStr;
+}
+
+void TimerDialog::decrementTime()
+{
+    if (started) {
+        if (time == 60 && this->isHidden()) //show timers a minute before they go off
+            this->show();
+        else if (time == 0)
+            timesUp();
+        else {
+            time = time - 1;
+            updateTime();
+        }
+    }
+}
+
+void TimerDialog::on_setSoundButton_clicked()
+{
+    //Taken form old brewtarget timers
+    QDir soundsDir = QString("%1sounds/").arg(Brewtarget::getDataDir());
+    QString soundFile = QFileDialog::getOpenFileName( qobject_cast<QWidget*>(this), tr("Open Sound"), soundsDir.exists() ? soundsDir.canonicalPath() : "", tr("Audio Files (*.wav *.ogg *.mp3 *.aiff)") );
+
+    if( soundFile.isNull() )
+    {
+       Brewtarget::logW("Null sound file.");
+       return;
+    }
+    setSound(soundFile);
+    // Indicate a sound is loaded
+    setSoundButton->setCheckable(true);
+    setSoundButton->setChecked(true);
+}
+
+void TimerDialog::setDefualtAlarmSound()
+{
+    QString soundFile = QString("%1sounds/").arg(Brewtarget::getDataDir()) + "beep.ogg";
+    setSound(soundFile);
+}
+
+void TimerDialog::setSound(QString s)
+{
+    //Taken from old brewtarget timers
+#ifndef NO_QTMULTIMEDIA
+   if( !playlist->clear() )
+      Brewtarget::logW(playlist->errorString());
+   if( !playlist->addMedia(QUrl::fromLocalFile(s)) )
+      Brewtarget::logW(playlist->errorString());
+   playlist->setCurrentIndex(0);
+#endif
+}
+
+void TimerDialog::on_setTimeBox_valueChanged(int t)
+{
+    if (t*60 > boilTime->getTime()) {
+        QMessageBox::warning(this, "Error", "Addition time cannot be longer than remaining boil time");
```

Comment:
Aah, good spotting. Will sort that out.
On 15 Dec 2015 15:43, "theophae" <notifications@github.com> wrote:

> In src/TimerDialog.cpp
> <https://github.com/Brewtarget/brewtarget/pull/122#discussion_r47636615>:
>
> > +void TimerDialog::setSound(QString s)
> > +{
> > +    //Taken from old brewtarget timers
> > +#ifndef NO_QTMULTIMEDIA
> > +   if( !playlist->clear() )
> > +      Brewtarget::logW(playlist->errorString());
> > +   if( !playlist->addMedia(QUrl::fromLocalFile(s)) )
> > +      Brewtarget::logW(playlist->errorString());
> > +   playlist->setCurrentIndex(0);
> > +#endif
> > +}
> > +
> > +void TimerDialog::on_setTimeBox_valueChanged(int t)
> > +{
> > +    if (t*60 > boilTime->getTime()) {
> > +        QMessageBox::warning(this, "Error", "Addition time cannot be longer than remaining boil time");
>
> Missing tr()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Brewtarget/brewtarget/pull/122/files#r47636615>.
>